src/redis_cmd.cc

```diff
@@ -4194,184 +4047,184 @@ struct redisCommand {
-    ADD_CMD("publish", 0, 0, 0, CommandPublish),
-    ADD_CMD("subscribe", 0, 0, 0, CommandSubscribe),
-    ADD_CMD("unsubscribe", 0, 0, 0, CommandUnSubscribe),
-    ADD_CMD("psubscribe", 0, 0, 0, CommandPSubscribe),
-    ADD_CMD("punsubscribe", 0, 0, 0, CommandPUnSubscribe),
-    ADD_CMD("pubsub", 0, 0, 0, CommandPubSub),
+    ADD_CMD("publish", 3, false, 0, 0, 0, CommandPublish),
+    ADD_CMD("subscribe", -2, false, 0, 0, 0, CommandSubscribe),
+    ADD_CMD("unsubscribe", -1, false, 0, 0, 0, CommandUnSubscribe),
+    ADD_CMD("psubscribe", -2, false, 0, 0, 0, CommandPSubscribe),
+    ADD_CMD("punsubscribe", -1, false, 0, 0, 0, CommandPUnSubscribe),
+    ADD_CMD("pubsub", -2, false, 0, 0, 0, CommandPubSub),
 
     // Sortedint command
-    ADD_CMD("siadd", 1, 1, 1, CommandSortedintAdd),
-    ADD_CMD("sirem", 1, 1, 1, CommandSortedintRem),
-    ADD_CMD("sicard", 1, 1, 1, CommandSortedintCard),
-    ADD_CMD("siexists", 1, 1, 1, CommandSortedintExists),
-    ADD_CMD("sirange", 1, 1, 1, CommandSortedintRange),
-    ADD_CMD("sirevrange", 1, 1, 1, CommandSortedintRevRange),
-    ADD_CMD("sirangebyvalue", 1, 1, 1, CommandSortedintRangeByValue),
-    ADD_CMD("sirevrangebyvalue", 1, 1, 1, CommandSortedintRevRangeByValue),
+    ADD_CMD("siadd", -3, true, 1, 1, 1, CommandSortedintAdd),
+    ADD_CMD("sirem", -3, true, 1, 1, 1, CommandSortedintRem),
+    ADD_CMD("sicard", 2, false, 1, 1, 1, CommandSortedintCard),
+    ADD_CMD("siexists", -3, false, 1, 1, 1, CommandSortedintExists),
+    ADD_CMD("sirange", -4, false, 1, 1, 1, CommandSortedintRange),
+    ADD_CMD("sirevrange", -4, false, 1, 1, 1, CommandSortedintRevRange),
+    ADD_CMD("sirangebyvalue", -4, false, 1, 1, 1, CommandSortedintRangeByValue),
+    ADD_CMD("sirevrangebyvalue", -4, false, 1, 1, 1, CommandSortedintRevRangeByValue),
 
     // internal management cmd
-    ADD_CMD("compact", 0, 0, 0, CommandCompact),
-    ADD_CMD("bgsave", 0, 0, 0, CommandBGSave),
-    ADD_CMD("flushbackup", 0, 0, 0, CommandFlushBackup),
-    ADD_CMD("slaveof", 0, 0, 0, CommandSlaveOf),
-    ADD_CMD("stats", 0, 0, 0, CommandStats),
+    ADD_CMD("compact", 1, false, 0, 0, 0, CommandCompact),
+    ADD_CMD("bgsave", 1, false, 0, 0, 0, CommandBGSave),
+    ADD_CMD("flushbackup", 1, false, 0, 0, 0, CommandFlushBackup),
+    ADD_CMD("slaveof", 3, false, 0, 0, 0, CommandSlaveOf),
+    ADD_CMD("stats", 1, false, 0, 0, 0, CommandStats),
 
     // Replicaion commands
-    ADD_CMD("replconf", 0, 0, 0, CommandReplConf),
-    ADD_CMD("psync", 0, 0, 0, CommandPSync),
-    ADD_CMD("_fetch_meta", 0, 0, 0, CommandFetchMeta),
-    ADD_CMD("_fetch_file", 0, 0, 0, CommandFetchFile),
-    ADD_CMD("_db_name", 0, 0, 0, CommandDBName),
+    ADD_CMD("replconf", -3, false, 0, 0, 0, CommandReplConf),
+    ADD_CMD("psync", 2, false, 0, 0, 0, CommandPSync),
+    ADD_CMD("_fetch_meta", 1, false, 0, 0, 0, CommandFetchMeta),
+    ADD_CMD("_fetch_file", 2, false, 0, 0, 0, CommandFetchFile),
+    ADD_CMD("_db_name", 1, false, 0, 0, 0, CommandDBName),
```

Comment:
for `is_write`, we can use one description，such as "write, ok-loading, access-db, lock-db", and we parse it generate its flags, it is much more readable.

Reply:
yeah, it's more readable with literal but I think can improve it at the next PR, to keep this PR more simple and clear by just move command attributes together.